packages/react-combobox/Spec.md

```diff
@@ -2,62 +2,515 @@
+  </OptionGroup>
+  <OptionGroup label="Group 2">
+    <Option key="C">Option C</Option>
+    <Option key="D">Option D</Option>
+  </OptionGroup>
+</Combobox>
+```
+
+### DOM
+
+This shows the DOM structure of the Combobox after being opened:
+
+```html
+<div>
+  <!-- root slot, combobox wrapper -->
+  <button role="combobox" type="button" aria-expanded="true" aria-activedescendant="option1-id">
+    <!-- primary slot -->
+    Select an option
+  </button>
+  <span
+    ><svg><!-- dropdown icon --></svg></span
+  >
+</div>
+
+<!-- in a portal: -->
+<div role="listbox">
+  <!-- listbox root slot -->
+  <div role="group" aria-labelledby="group1-label-id">
+    <!-- optiongroup root slot -->
+    <span id="group1-label-id" role="presentation">Group 1</span
+    ><!-- optiongroup label slot -->
+    <div role="option" aria-selected="false" id="option1-id">
+      <!-- option root slot -->
+      <span aria-hidden="true"
+        ><svg><!-- check icon --></svg></span
+      ><!-- option check slot -->
+      Option A
+    </div>
+    <div role="option" aria-selected="false" id="option2-id">
+      <span aria-hidden="true"
+        ><svg><!-- check icon --></svg></span
+      >
+      Option B
+    </div>
+  </div>
+</div>
+```
 
 ## Migration
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+The following v7/v8 and v0 components can be migrated to the v9 Combobox:
+
+- **v8 Dropdown**: Use the v9 Select or v9 Combobox, [depending on the required features](#combobox-vs-select-vs-menu). If using Combobox, use the [select-only variation](#select-only-combobox).
+- **v8 Combobox**: The v8 Combobox should be replaced by the editable variant of the v9 Combobox.
+- **v8 Pickers**: v8 Pickers should be replaced by the multiselect variant of the v9 Combobox. There is a design spec for adding selected tags to the v9 Combobox.
+- **v0 Dropdown**: The v0 Dropdown should be replaced by the v9 Combobox.
+
+The primary difference between v7/v8 and v0 components vs. the v9 Combobox is the definition of options as children in v9.
 
-- _Migration from v8_
-- _Migration from v0_
+For a full migration guide, see [the migration spec](./Spec-migration.md).
 
 ## Behaviors
 
-_Explain how the component will behave in use, including:_
+### Positioning
 
-- _Component States_
-- _Interaction_
-  - _Keyboard_
-  - _Cursor_
-  - _Touch_
-  - _Screen readers_
+The default position for the listbox popup is `below-start`. The position can be customized through the `positioning` prop, which shares an API with other v9 components like Menu and Tooltip. The full positioning API is defined in the [`@fluentui/react-positioning` package](https://github.com/microsoft/fluentui/blob/master/packages/react-positioning/src/types.ts).
+
+### Keyboard interaction
+
+The keyboard interaction model follows that of the [ARIA Practices Combobox Pattern](https://w3c.github.io/aria-practices/#combobox), with the addition of <kbd>Space</kbd> to select options.
```

Comment:
when we implemented v0 Dropdown, we also had to add special handling for Tab - the requirement was that Tab should also select the active item from the list. This was not in the ARIA pattern, however major comboboxes in MS products do that, so users are used to the behavior.

Reply:
Yup, I definitely would like this to happen too -- assuming you mean tabbing away from an open dropdown selects the currently highlighted item?

Good call, I'll add this to the spec.

Reply:
yes, that is what I meant. thank you